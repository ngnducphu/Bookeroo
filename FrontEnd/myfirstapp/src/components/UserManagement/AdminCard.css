.shopowner-card {
  height: 22rem;
  margin-left: 2%;
  margin-right: 2%;
}

.approve-btn {
  margin-right: 2%;
}
