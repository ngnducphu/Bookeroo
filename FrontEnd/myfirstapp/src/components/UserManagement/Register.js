import React, { Component } from "react";
import { createNewUser } from "../../actions/securityActions";
import * as PropTypes from "prop-types";
import { ToastContainer } from "react-toastify";

import { connect } from "react-redux";

class Register extends Component {
  constructor() {
    super();

    this.state = {
      username: "",
      password: "",
      confirmPassword: "",
      email: "",
      firstname: "",
      lastname: "",
      address: "",
      phoneNumber: "",
      usertype: "public",
      abnnumber: "",
      imageUrl: "",
    };
    this.onChange = this.onChange.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  // new user state object

  onSubmit(e) {
    e.preventDefault();

    const newUser = {
      username: this.state.username,
      password: this.state.password,
      confirmPassword: this.state.confirmPassword,
      givenName: this.state.firstname,
      surname: this.state.lastname,
      address: this.state.address,
      phonenumber: this.state.phonenumber,
      roleCode: "",
      abn: this.state.abnnumber,
    };
    if (this.state.usertype === "shop owner") {
      newUser.roleCode = "SHOP";
    } else if (this.state.usertype === "admin") {
      newUser.roleCode = "ADMIN";
    } else if (this.state.usertype === "public") {
      newUser.roleCode = "PUBLIC";
    }
    this.props.createNewUser(newUser, this.props.history);
  }
  /*  on clicking submit a neUser object is created from details entered
     and the createNewUser in securityAction is called passing the neUser and history as a parameter */

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
    this.setState({ abnnumber: "" });
  }

  render() {
    return (
      <div className="register">
        <div className="container">
          <ToastContainer position="top-center" autoClose={2000} />

          <div className="row">
            <div className="col-md-8 m-auto">
              <h1 className="display-4 text-center">Sign Up</h1>
              <p className="lead text-center">Create your Account</p>
              <form onSubmit={this.onSubmit}>
                <div className="form-group">
                  <input
                    type="text"
                    className="form-control form-control-lg"
                    placeholder="Username"
                    name="username"
                    value={this.state.username}
                    onChange={this.onChange}
                  />
                </div>
                <div className="form-group">
                  <input
                    type="text"
                    className="form-control form-control-lg"
                    placeholder="First Name"
                    name="firstname"
                    value={this.state.firstname}
                    onChange={this.onChange}
                  />
                </div>
                <div className="form-group">
                  <input
                    type="text"
                    className="form-control form-control-lg"
                    placeholder="Last Name"
                    name="lastname"
                    value={this.state.lastname}
                    onChange={this.onChange}
                  />
                </div>
                <div className="form-group">
                  <input
                    type="text"
                    className="form-control form-control-lg"
                    placeholder="Address"
                    name="address"
                    value={this.state.address}
                    onChange={this.onChange}
                  />
                </div>
                <div className="form-group">
                  <input
                    type="text"
                    className="form-control form-control-lg"
                    placeholder="Phone number"
                    name="phoneNumber"
                    value={this.state.phonenumber}
                    onChange={this.onChange}
                  />
                </div>
                <div className="form-group">
                  <select
                    defaultValue={this.state.selectValue}
                    onChange={this.handleChange}
                    name="usertype"
                    className="form-control form-control-lg"
                  >
                    <option value="public">Public User</option>
                    <option value="shop owner">Shop owner</option>
                  </select>
                </div>

                {this.state.usertype === "shop owner" && (
                  <div className="form-group">
                    <input
                      type="text"
                      className="form-control form-control-lg"
                      placeholder="ABN Number"
                      name="abnnumber"
                      value={this.state.abnnumber}
                      onChange={this.onChange}
                    />
                  </div>
                )}

                <div className="form-group">
                  <input
                    type="password"
                    className="form-control form-control-lg"
                    placeholder="Password"
                    name="password"
                    value={this.state.password}
                    onChange={this.onChange}
                  />
                </div>

                <div className="form-group">
                  <input
                    type="password"
                    className="form-control form-control-lg"
                    placeholder="Confirm Password"
                    name="confirmPassword"
                    value={this.state.confirmPassword}
                    onChange={this.onChange}
                  />
                </div>
                <input type="submit" className="btn btn-info btn-block mt-4" />
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

/* prop validator to ensure warnings are shown whenever prop is not provided */
Register.propTypes = {
  createNewUser: PropTypes.func.isRequired,
  errors: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  errors: state.errors,
});
export default connect(mapStateToProps, { createNewUser })(Register);

export const RegisterPlain = (props) => {
  return (
    <div className="register">
      <div className="container">
        <div className="row">
          <div className="col-md-8 m-auto">
            <h1 className="display-4 text-center">Sign Up</h1>
            <p className="lead text-center">Create your Account</p>
            <form onSubmit={props.onSubmit}>
              <div className="form-group">
                <input
                  type="text"
                  className="form-control form-control-lg"
                  placeholder="Username"
                  name="username"
                />
              </div>
              <div className="form-group">
                <input
                  type="text"
                  className="form-control form-control-lg"
                  placeholder="First Name"
                  name="firstname"
                />
              </div>
              <div className="form-group">
                <input
                  type="text"
                  className="form-control form-control-lg"
                  placeholder="Last Name"
                  name="lastname"
                />
              </div>
              <div className="form-group">
                <input
                  type="text"
                  className="form-control form-control-lg"
                  placeholder="Address"
                  name="address"
                />
              </div>
              <div className="form-group">
                <input
                  type="text"
                  className="form-control form-control-lg"
                  placeholder="Phone number"
                  name="phoneNumber"
                />
              </div>
              <div className="form-group">
                <select
                  name="usertype"
                  className="form-control form-control-lg"
                >
                  <option value="public">Public User</option>
                  <option value="shop owner">Shop owner</option>
                </select>
              </div>

              <div className="form-group">
                <input
                  type="text"
                  className="form-control form-control-lg"
                  placeholder="ABN Number"
                  name="abnnumber"
                />
              </div>

              <div className="form-group">
                <input
                  type="password"
                  className="form-control form-control-lg"
                  placeholder="Password"
                  name="password"
                />
              </div>

              <div className="form-group">
                <input
                  type="password"
                  className="form-control form-control-lg"
                  placeholder="Confirm Password"
                  name="confirmPassword"
                />
              </div>
              <input type="submit" className="btn btn-info btn-block mt-4" />
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};
