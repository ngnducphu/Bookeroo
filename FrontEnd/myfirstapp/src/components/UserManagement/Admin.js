import React, { useEffect } from "react";
import AdminCard from "./AdminCard";
import { connect } from "react-redux";
import { ToastContainer } from "react-toastify";
import {
  getPendingShopOwners,
  approveShopOwner,
} from "../../actions/personActions";

const Admin = ({
  pendingShopOwners,
  approveShopOwner,
  getPendingShopOwners,
}) => {
  useEffect(() => {
    getPendingShopOwners();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div>
      <ToastContainer position="top-center" autoClose={2000} />
      {pendingShopOwners.length === 0 ? (
        <div style={{ paddingLeft: "5%", paddingTop: "2%" }}>
          <h1>No Pending Shop Owners</h1>
        </div>
      ) : (
        pendingShopOwners.map((pendingShopOwner, index) => {
          return (
            <div key={index}>
              <AdminCard
                shopOwner={pendingShopOwner}
                approveShopOwner={approveShopOwner}
              />
            </div>
          );
        })
      )}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    pendingShopOwners: state.person.pendingShopOwners,
  };
};
export default connect(mapStateToProps, {
  getPendingShopOwners,
  approveShopOwner,
})(Admin);
