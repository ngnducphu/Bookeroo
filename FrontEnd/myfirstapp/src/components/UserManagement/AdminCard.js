import React from "react";
import "./AdminCard.css";

const AdminCard = ({ shopOwner, approveShopOwner }) => {
  const handleApprove = () => {
    approveShopOwner(shopOwner.id);
  };

  return (
    <div className="card shopowner-card">
      <h5 className="card-header">
        Shop Owner Name: {shopOwner.givenName} {shopOwner.surname}
      </h5>
      <div className="card-body">
        <h5 className="card-title">Details</h5>
        <p className="card-text">
          <b>Email Address:</b> {shopOwner.username}
        </p>
        <p className="card-text">
          <b>ABN: </b>
          {shopOwner.abn}
        </p>
        <p className="card-text">
          <b>Address: </b> {shopOwner.address}
        </p>
        <p className="card-text">
          {" "}
          <b>Phone Number:</b> {shopOwner.phoneNumber}
        </p>
        <button className="btn btn-primary approve-btn" onClick={handleApprove}>
          Approve
        </button>
        <button className="btn btn-danger reject-btn">Reject</button>
      </div>
    </div>
  );
};

export default AdminCard;
