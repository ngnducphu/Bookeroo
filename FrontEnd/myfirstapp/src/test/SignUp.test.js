import React from "react";
import { render, cleanup } from "@testing-library/react";
import { RegisterPlain } from "../components/UserManagement/Register";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);

test("Render without crashing", () => {
  render(
    <BrowserRouter>
      <RegisterPlain />
    </BrowserRouter>
  );
});

test("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <RegisterPlain />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
