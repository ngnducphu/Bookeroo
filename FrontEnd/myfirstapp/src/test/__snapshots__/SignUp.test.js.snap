// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="register"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-8 m-auto"
      >
        <h1
          className="display-4 text-center"
        >
          Sign Up
        </h1>
        <p
          className="lead text-center"
        >
          Create your Account
        </p>
        <form>
          <div
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              name="username"
              placeholder="Username"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              name="firstname"
              placeholder="First Name"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              name="lastname"
              placeholder="Last Name"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              name="address"
              placeholder="Address"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              name="phoneNumber"
              placeholder="Phone number"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <select
              className="form-control form-control-lg"
              name="usertype"
            >
              <option
                value="public"
              >
                Public User
              </option>
              <option
                value="shop owner"
              >
                Shop owner
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              name="abnnumber"
              placeholder="ABN Number"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              name="password"
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              name="confirmPassword"
              placeholder="Confirm Password"
              type="password"
            />
          </div>
          <input
            className="btn btn-info btn-block mt-4"
            type="submit"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
